File: tests/sources/basic/02-basicTasks/src/main/java/basicTasks/Simple.java
package basicTasks;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;


public class Simple {

    public static void main(String[] args) {
        // Check and get parameters
        if (args.length != 1) {
            System.out.println("[ERROR] Bad number of parameters");
            System.out.println("    Usage: simple.Simple <counterValue>");
            System.exit(-1);
        }
        String counterName = "counter";
        int initialValue = Integer.parseInt(args[0]);

        // Write value
        try {
            FileOutputStream fos = new FileOutputStream(counterName);
            fos.write(initialValue);
            System.out.println("Initial counter value is " + initialValue);
            fos.close();
        } catch (IOException ioe) {
            ioe.printStackTrace();
            System.exit(-1);
        }

        // ------------------------------------------------------------------------
        // Execute increment
        SimpleImpl.increment(counterName);
        // Read new value
        System.out.println("After Sending task");
        String test = "Name= " + counterName;
        System.out.println("In method Checking result " + test);
        try {
            FileInputStream fis = new FileInputStream(counterName);
            System.out.println("Final counter value is " + fis.read());
            fis.close();
        } catch (IOException ioe) {
            ioe.printStackTrace();
            System.exit(-1);
        }

        // Execute increment
        SimpleImpl.increment(counterName);
        // Read from private method
        checkResult(counterName);

        // Execute increment
        SimpleImpl.increment(counterName);
        // Read from blackbox method
        Reader.checkResult(counterName);
    }

    private static void checkResult(String counterName) {
        System.out.println("Private Checking result");
        try {
            FileInputStream fis = new FileInputStream(counterName);
            System.out.println("Final counter 2 value is " + fis.read());
            fis.close();
        } catch (IOException ioe) {
            ioe.printStackTrace();
            System.exit(-1);
        }
    }

}

File: tests/sources/basic/27-simple_python/src/simple.py
#!/usr/bin/python

# -*- coding: utf-8 -*-

"""
PyCOMPSs Testbench Simple
========================
"""

# Imports
from pycompss.api.parameter import *
from pycompss.api.task import task
import unittest
import sys


@task(filePath=FILE_INOUT)
def increment(filePath):
    print("Init task user code")
    # Read value
    fis = open(filePath, 'r')
    value = fis.read()
    print("Received " + value)
    fis.close()

    # Write value
    fos = open(filePath, 'w')
    newValue = str(int(value) + 1)
    print("Computed " + newValue)
    fos.write(newValue)
    fos.close()


def usage():
    print("[ERROR] Bad number of parameters")
    print("    Usage: simple <counterValue>")


class KmeansTest(unittest.TestCase):
    def test_simple(self):
        from pycompss.api.api import compss_open


        initial_value = '1'
        fileName = "counter"

        # Write value
        fos = open(fileName, 'w')
        fos.write(initial_value)
        fos.close()
        print("Initial counter value is " + initial_value)

        # Execute increment
        increment(fileName)

        # Write new value
        fis = compss_open(fileName, 'r+')
        final_value = fis.read()
        fis.close()
        print("Final counter value is " + final_value)

        expected_final_value = '2'
        if final_value != expected_final_value:
            print("Simple increment test result is incorrect. "
                  "Expected: %s, got: %s" % (expected_final_value, final_value))
        self.assertEqual(final_value, expected_final_value)


if __name__ == '__main__':
    unittest.main()
